ecma402_traits/src/pluralrules.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Contains the API configuration as prescribed by [ECMA 402][ecma].
+///
+///    [ecma]: https://www.ecma-international.org/publications/standards/Ecma-402.htm
+///
+/// The meaning of the options is the same as in the similarly named
+/// options in the JS version.
+///
+/// See [Options] for the contents of the options.  See the [PluralRules::try_new]
+/// for the use of the options.
+pub mod options {
+    /// The type of plural rules with respect to cardinality and ordinality.
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub enum Type {
+        /// As in "one, two, three, ..."
+        Cardinal,
+        /// As in: "first, second, third, ..."
+        Ordinal,
+    }
+}
+
+/// The options set by the user at construction time.  See discussion at the top level
+/// about the name choice.  Provides as a "bag of options" since we don't expect any
+/// implementations to be attached to this struct.
+///
+/// The default values of all the options are prescribed in by the [TC39 report][tc39lf].
+///
+///   [tc39lf]: https://tc39.es/proposal-intl-list-format/#sec-Intl.ListFormat
+#[derive(Eq, PartialEq, Debug, Clone)]
+pub struct Options {
+    /// Selects a [options::Type] for the formatted list.  If unset, defaults to
+    /// [options::Type::Cardinal].
+    pub in_type: options::Type,
+    pub minimum_integer_digits: u8,
+    pub minimum_fraction_digits: u8,
+    pub maximum_fraction_digits: u8,
+    pub minimum_significant_digits: u8,
+    pub maximum_significant_digits: u8,
```

Comment:
Would it make sense to consider using `std::ops::RangeBounds`? 

I feel like it makes it a bit more intuitive that we're talking ranges, rather than independent min/max values. But I'm not sure how traits will work here, so if that's a no-go, then the current layout is ok.

Reply:
The idea of using a dedicated, already-existing type is appealing.  

I noticed that it's not obvious which is the concrete type implementing `RangeBounds` (it's a trait).  Also, it has a whole lot of complication with the use of inclusive and exclusive bounds, for which we don't have the need here.

Long story short, I'm not quite seeing how to get the benefit of a dedicated type without complicating the trait definition too much.  Given that, I'd opt to keep things as they are for the time being.  I'm open to reworking the options if something better comes along.

And of course, I may simply be wrong in the above assessment, in which case I welcome corrections.

Reply:
yeah, we can revisit it later :)